### ISSUE TITLE
Communicator App not sending messages

### ISSUE THREAD

**Issue by Pkoli81** [OfficeDev/microsoft-teams-apps-company-communicator#720]
![CCPic_Issue](https://user-images.githubusercontent.com/105282162/167604875-83d78f50-654a-480c-9df0-12a433c1e874.png)

Company Communicator not sending messages,
stuck at sending status

Regards,
Praveen
+65901822299

**Comment by gsv022** [comment]
Hi @Pkoli81 ,
The above sent picture seems to be related to activity logs.
Could you please share the screenshot of status when the message is stuck(UI view)? It will be helpful for better troubleshooting from our end.

**Comment by Pkoli81** [comment]
Hi Sai Venkat,
Really appreciate you reaching out, below is the screen shot from the Teams UI, please let me know if more information is required.

![image](https://user-images.githubusercontent.com/105282162/167860651-0c52b6ca-7274-4603-adf9-2481c4530bab.png)

**Comment by gsv022** [comment]
Hi again,
Could you please cross verify all the 3 function are successfully deployed? And restart all the functions and try for a message again.

**Comment by Pkoli81** [comment]
Sure Sai, let me try that and get back to you.

**Comment by Pkoli81** [comment]
HI Team/Sai, i have doubly checked the 3 functions, and its all as per the article, however the issue remains the same.
Not sure how to go ahead next? 

I seriously need some help here to get this fixed.

**Comment by wendyhaddad** [comment]
I ran into the same issue.  Checking the 3 functions did not work for me.  Here's what wound up working for me:

1.) Open the Resource group that you created
2.) Click the "Overview" tab and select "Deployments" in the top right of the screen.  You should see something that says how many failed or succeeded
3.) Click on the Microsoft.Template-2022042... link that failed
4.) Select redeploy

Hope that helps you!

**Comment by Pkoli81** [comment]
Any Idea team? possible to get any working code please? i need to get this fixed ASAP..Will really really appreciate if someone could help here.

**Comment by cristianoag** [comment]
@Pkoli81 - Try cleaning the npm cache on your deployment and then resync. Follow the steps below:

1. Open portal.azure.com and use an admin credential to authenticate.
2. Navigate to the resource group where all CC resources are installed.
3. Click the App Service resource.
4. On the left rail, scroll down to Development Tools and click Advanced Tools.
5. Click Go
6. Now on the menu located on the top of the page, select Debug console, and then click CMD
7. Navigate to site\repository\Source\CompanyCommunicator\ClientApp. You can do that clicking on the folders on the box located on the top of the page or using the CD command on the MS-DOS command console
8. Delete the node_modules folder by running **DEL /S /Q node_modules** and then **RD /S /Q node_modules**
9. Delete the package-lock.json file
10. Now clear the cache, you need to run the npm cache clean --force command in terminal.
11. Run the npm install command in the terminal. (This will take some time).
12. When finished, retry the sync process, and let me know the results.

**Comment by Pkoli81** [comment]
Thank you Cristian, i have performed the steps as suggested

![image](https://user-images.githubusercontent.com/105282162/169452803-880b0243-23ba-4ab7-944e-101129f277d6.png)
![image](https://user-images.githubusercontent.com/105282162/169452814-44222875-2c86-423d-abdc-2f19befa53f4.png)

Failed again with below error,
![image](https://user-images.githubusercontent.com/105282162/169452871-2bd57cb8-1457-4a5e-a9cd-1302bf771fdd.png)

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
Input string was not in a correct format.

**Comment by cristianoag** [comment]
Ok. Can you please go back to the debug console and execute: 
npm i fsevents@latest -f --save-optional

Then resync. That should solve the issue.

**Comment by Pkoli81** [comment]
Hi Cristian, it failed again with below error,

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
Input string was not in a correct format.

**Comment by Pkoli81** [comment]
Any further ideas team? possible to get any previous version of the code plz? im running out of time, i need to show this demo to the customer ASAP

**Comment by Pkoli81** [comment]
Dear Team,
Still waiting to find some solution for this issue? i have been asking for previous version code, if this one has got issues, possible for anyone to share, please?

**Comment by gsv022** [comment]
Hi @Pkoli81 ,
The fsevents library is causing the app service deployment to fail. We are working on the fix and will keep you updated. To unblock the deployment please follow the below steps as a quick workaround,

- Go to portal.azure.com. Navigate to resource group where all CC resources are deployed.
- Click on the App service -> Click on Configuration
- Click on **WEBSITE_NODE_DEFAULT_VERSION.**
- Update the default value to **16.13.0** (previous value ->10.15.2)
- Click on Overview and Re-Start the app service.
- Once the app service is restarted. Navigate to **Deployment Center** and click on Sync.
- Wait till the deployment is completed. You can validate this once the status changes to **Success** under logs. 


![image](https://user-images.githubusercontent.com/98143492/170511053-7b2a0cb8-7bc4-477e-b6a6-359012cf6235.png)

![image](https://user-images.githubusercontent.com/98143492/170511111-31010f41-e065-4cd3-a548-075bc61821c3.png)

- For azure function failure, please refer this troubleshooting guide [issue-1] (https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/wiki/Troubleshooting)

**Comment by Pkoli81** [comment]
Awesome thanks Sai, so far so good, it crossed the last failures, will update how it goes from here.

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})

> company-communicator@5.1.0 build C:\home\site\repository\Source\CompanyCommunicator\ClientApp
> react-scripts build

Creating an optimized production build...

**Comment by v-royavinash** [comment]
@Pkoli81 Can you please share your MS teams id?

**Comment by Pkoli81** [comment]
Sorry however it failed again

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
Input string was not in a correct format.

**Comment by Pkoli81** [comment]
> 
MS Teams ID means?

**Comment by v-royavinash** [comment]
Can we connect over Microsoft Teams for a call to troubleshoot the issue? We wanted to double check the steps performed and the logs generated.

**Comment by Pkoli81** [comment]
HI JasthiKalyani, yes user app is uploaded at the tenant level and auth is in place as per the documentation, what is proactive installation enablement?

![image](https://user-images.githubusercontent.com/105282162/171337611-3b1d4df0-dd9a-4788-8e1b-a4c13f6d586e.png)

**Comment by Pkoli81** [comment]
Dear Team, thanks to @v-royavinash, we managed to solve this issue.

Now the only concern is, I want to give this Company communicator Demo for our Government organization here in Singapore, they have 100s of agencies (Domains) in the same tenant, how will the governance work so that an author can post the messages for their respective domains? is multi-domain single-tenant supported for CC App? How to assign permissions for all such domains?

**Comment by Pkoli81** [comment]
Hi Team, any idea about the last question, apart from creating dynamic DLs I don't see any other option to manage permissions for the multi-domain environment for the CC app?

**Comment by SankarMicrosoft** [comment]
Please let us know if you are still facing this issue and require assistance and we will be happy to discuss. If no further action is needed at moment and no response is received, we will close out the issue in 3 business days. You may also always open a new issue or respond here if something does come back up.

**Comment by maye-vargas** [comment]
@v-royavinash Can you please tell me the solution you gave for the error, please, it is that I am presenting the same error, but in my case they are sent but they do not reach the users of the equipment.

**Comment by Sandeep-FED** [comment]
> Dear Team, thanks to @v-royavinash, we managed to solve this issue.
> 
> Now the only concern is, I want to give this Company communicator Demo for our Government organization here in Singapore, they have 100s of agencies (Domains) in the same tenant, how will the governance work so that an author can post the messages for their respective domains? is multi-domain single-tenant supported for CC App? How to assign permissions for all such domains?

Hi @Pkoli81 How you resolved this issue?

**Comment by cristianoag** [comment]
You can customize the code or use one of the variations modified by the community that have that feature.
